### ISSUE TITLE
assinging lengh on a buffer?

### ISSUE THREAD

**Issue by apolishch** [SheetJS/js-xlsx#313]
I am experiencing the following error:

Cannot set property length of [object Object] which has only a getter
<     at sbcs_d (/Users/abraham/env/dev/import/node_modules/xlsx/dist/cpexcel.js:909:20)
<     at Object.decode (/Users/abraham/env/dev/import/node_modules/xlsx/dist/cpexcel.js:1198:40)
<     at _gc2 (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:28:24)
<     at Buffer.ReadShift [as read_shift](/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:1629:13)
<     at parse_ShortXLUnicodeString (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:3088:21)
<     at Object.parse_Font [as f](/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:3476:13)
<     at slurp (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:9235:11)
<     at parse_workbook (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:9356:15)
<     at parse_xlscfb (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:9833:27)
<     at readSync (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:11397:11)
<     at Object.readFileSync (/Users/abraham/env/dev/import/node_modules/xlsx/xlsx.js:11409:9)
<     at /Users/abraham/env/dev/import/routers/import.js:103:21
<     at tryCatcher (/Users/abraham/env/dev/import/node_modules/bluebird/js/main/util.js:26:23)
<     at Promise.module.exports.Promise._settlePromiseFromHandler (/Users/abraham/env/dev/import/node_modules/bluebird/js/main/promise.js:505:31)
<     at Promise.module.exports.Promise._settlePromiseAt (/Users/abraham/env/dev/import/node_modules/bluebird/js/main/promise.js:581:18)
<     at Promise.module.exports.Promise._settlePromises (/Users/abraham/env/dev/import/node_modules/bluebird/js/main/promise.js:697:14)

Looking at the line in question: `mdb.length = 2 * len;`

Which looks like its assigning a length on a Buffer?

Any insight into how this could happen would be appreciated...

**Comment by bryannaegele** [comment]
I'm seeing this as well. Node v.5.0.0.

There seems to be a fair amount of instances where the buffer length is attempted to be set. The docs warn against treating the length property on Buffer as it can create problems. The docs still say it's mutable, but apparently that may no longer the case. [buf.length](https://nodejs.org/api/buffer.html#buffer_buf_length)

![xlsx-error](https://cloud.githubusercontent.com/assets/5731285/11573758/95d73686-99cd-11e5-9b3b-4d25e5d7f504.png)

```javascript
var sbcs_decode = function make_sbcs_decode(cp) {
      var D = cpt[cp].dec;
      var DD = new Buffer(131072), d=0, c;
      for(d=0;d<D.length;++d) {
        if(!(c=D[d])) continue;
        var w = c.charCodeAt(0);
        DD[2*d] = w&255; DD[2*d+1] = w>>8;
      }
      return function sbcs_d(data) {
        var len = data.length, i=0, j;
        if(2 * len > mdl) { mdl = 2 * len; mdb = new Buffer(mdl); }
        if(Buffer.isBuffer(data)) {
          for(i = 0; i < len; i++) {
            j = 2*data[i];
            mdb[2*i] = DD[j]; mdb[2*i+1] = DD[j+1];
          }
        } else if(typeof data === "string") {
          for(i = 0; i < len; i++) {
            j = 2*data.charCodeAt(i);
            mdb[2*i] = DD[j]; mdb[2*i+1] = DD[j+1];
          }
        } else {
          for(i = 0; i < len; i++) {
            j = 2*data[i];
            mdb[2*i] = DD[j]; mdb[2*i+1] = DD[j+1];
          }
        }
        mdb.length = 2 * len;
        return mdb.toString('ucs2');
      };
    };
```

I have a fix and will file a PR.

**Comment by psawaya** [comment]
Been running into this problem as well. Why wasn't #331 merged?

**Comment by bryannaegele** [comment]
@psawaya The Travis build was broken. There was a chain of issues in [js-codepage](https://github.com/SheetJS/js-codepage) that had to be resolved first which only @SheetJSDev could correct which has been done in https://github.com/SheetJS/js-codepage/commit/8bc8da47bafccc2bcb05876af355615fca385b28. 

I just tried bumping it again but the tests are still broken on newer versions of node. There's more work to be done to get this repo updated. Unfortunately, I don't have time to investigate further at this time. Hopefully @SheetJSDev can provide some insight for you.

**Comment by robgraeber** [comment]
+1

**Comment by maxjanisse** [comment]
Complete newb here but I ran into this problem on one of my projects.  There's bound to be a better way of fixing this for everyone else but here's what I did:

1. Updated `packages.json` to pull in codepage v1.4.0
2. Ran `npm install` to download the new files
3. Copied the `cpexcel.full.js` and `cputils.js` files from codepage into the ~/dist folder of xlsx
4. Renamed `cpexcel.js` to `cpexcel.js.bak` and `cpexcel.full.js` to `cpcexcel.js`

After that roundabout patching process, I no longer see this error. There is obviously a better way of doing this, but hopefully this helps.
